### ISSUE TITLE
[Bug]: TimeoutException after 0:00:10.000000: Future not completed

### ISSUE THREAD

**Issue by jenilAipxperts** [Baseflow/flutter-geolocator#1611]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Linux
- [ ] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

GetgetCurrentPosition  and use this sample code  userPositionData = await Geolocator.getCurrentPosition(
                locationSettings: LocationSettings(
                  accuracy: LocationAccuracy.high,
                  distanceFilter: 10,
                  timeLimit: const Duration(seconds: 10),
                ));  
                
                
  I don't want a limit I want to connect forcefully user's location in my app but on an Android Redmi(Android 13) device it will not get user's current location

### Expected results

want to connect in every device

### Actual results

I Dont want to timeout my app content force location tracking so for this issue my app was stuckdont

### Code sample

 userPositionData = await Geolocator.getCurrentPosition(
                locationSettings: LocationSettings(
                  accuracy: LocationAccuracy.high,
                  distanceFilter: 10,
                  timeLimit: const Duration(seconds: 10),
                ));

### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

[Upload media here]

</details>

### Version

geolocator: ^13.0.2

### Flutter Doctor output

 flutter --version
Flutter 3.24.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision dec2ee5c1f (12 days ago) • 2024-11-13 11:13:06 -0800
Engine • revision a18df97ca5
Tools • Dart 3.5.4 • DevTools 2.37.3

**Comment by aarican1** [comment]
I got this error in my app. The future has not been completed for a long time. If you fix the error, let me know.

**Comment by br-programmer** [comment]
I’m experiencing the same issue when running on the iOS simulator. The location request doesn't seem to work as expected. Has anyone found a workaround or a potential fix for this?

**Comment by jenilAipxperts** [comment]
> I’m experiencing the same issue when running on the iOS simulator. The location request doesn't seem to work as expected. Has anyone found a workaround or a potential fix for this?

Hello @br-programmer 
In iOS, I’m not facing any issues. Please verify it on a real device.

**Comment by TimHoogstrate** [comment]
@jenilAipxperts,

Please play around with the location settings, if you are testing indoor or outside would also generate different results. Try to lower the accuracy and distance filter first. 

Kind regards,

**Comment by jenilAipxperts** [comment]
@TimHoogstrate 
Still facing same issue i tried low and lowest both but still I am getting same error

**Comment by jenilAipxperts** [comment]
@TimHoogstrate Any Update?

**Comment by longzhenqi** [comment]
> @TimHoogstrate Any Update?
do you have WeChat？We can add friends to communicate

**Comment by elethiame** [comment]
Same issue.
My phone is IQOO 13 with android 15 from Vivo.

**Comment by iwpz** [comment]
Pixel 8 Android 14
Also can not get position with `await Geolocator.getCurrentPosition()`.
Without timeLimit, await this means forever waiting.

**Comment by lfclfc** [comment]
mee too

**Comment by iwpz** [comment]
After some test, GMS phone can't get position in China, after I solved "network connection" problem, it works.

**Comment by langyuxiansheng** [comment]
> 经过一番测试，GMS手机在中国无法定位，解决“网络连接”问题后，就可以了。

这个是怎么解决的呢，我测试过华为、小米、还有其它机型，貌似只要没有Google服务的都不行

**Comment by jupeng123** [comment]
I have the same situation as the author of this question

**Comment by Aliffter** [comment]
I have the same issue.  
On devices such as iPhone, Xiaomi, and Huawei, Geolocator.getCurrentPosition can return the longitude and latitude normally.
But on Vivo devices, the longitude and latitude cannot be obtained.

**Comment by Aliffter** [comment]
> > 经过一番测试，GMS手机在中国无法定位，解决“网络连接”问题后，就可以了。
> 
> 这个是怎么解决的呢，我测试过华为、小米、还有其它机型，貌似只要没有Google服务的都不行



> > 经过一番测试，GMS手机在中国无法定位，解决“网络连接”问题后，就可以了。
> 
> 这个是怎么解决的呢，我测试过华为、小米、还有其它机型，貌似只要没有Google服务的都不行

解决了吗？

**Comment by Aliffter** [comment]
@langyuxiansheng 
解决了吗？

**Comment by longzhenqi** [comment]
这是因为geo用的谷歌地图，国内访问不了，你试试翻墙看看，翻墙应该是没有问题的。




猫吃小鱼干
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Baseflow/flutter-geolocator"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年2月8日(星期六) 下午3:16
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Baseflow/flutter-geolocator] [Bug]: TimeoutException after 0:00:10.000000: Future not completed (Issue #1611)





 
@langyuxiansheng
 解决了吗？

**Comment by langyuxiansheng** [comment]
> [@langyuxiansheng](https://github.com/langyuxiansheng) 解决了吗？

没有  解决不了。。。 安卓端，暂时用的是百度地图的定位SDK来处理的。

**Comment by langyuxiansheng** [comment]
> 这是因为geo用的谷歌地图，国内访问不了，你试试翻墙看看，翻墙应该是没有问题的。
> 
> 
> 
> 
> 猫吃小鱼干
> ***@***.***
> 
> 
> 
> &nbsp;
> […](#)

但是正常用户谁会去翻墙呀。。。 大部分使用APP的用户可能都是普通用户，特别是国内的。

**Comment by sanyuegirl** [comment]
我也遇到了 小米红米不行 华为和iOS可以

**Comment by eifr** [comment]
Im getting this constantly on iOS on the second+ time opening the app. on the first time, the app prompts location permission. any other time it just hangs and if i add timeout, it fails with timeout error

**Comment by cngeru** [comment]
Adding `<uses-permission android:name="android.permission.INTERNET" />` did it for me

**Comment by smallooo** [comment]
In China on android phone, still not working. But use native project method it can get position.

**Comment by longzhenqi** [comment]
> In China on android phone, still not working. But use native project method it can get position.

Yes,only use native project method it can get position. in china

**Comment by longzhenqi** [comment]
> > 这是因为geo用的谷歌地图，国内访问不了，你试试翻墙看看，翻墙应该是没有问题的。
> > 猫吃小鱼干
> > _**@**_.***
> >  
> > […](#)
> 
> 但是正常用户谁会去翻墙呀。。。 大部分使用APP的用户可能都是普通用户，特别是国内的。

你只能使用安卓原生去写，然后传给flutter了。我们是海外的项目，在国内开发，所以才需要翻墙

**Comment by smallooo** [comment]
Any package recommend if need use in China, except baidu and Amap, those are two heavy, only need to get location.
